dbt/adapters/spark/connections.py

```diff
@@ -136,6 +108,10 @@ def execute(self, sql, bindings=None):
             ThriftState.FINISHED_STATE,
         ]
 
+        # Convert decimal.Decimal to float as PyHive doesn't work with decimals
+        if bindings:
+            bindings = [float(x) if isinstance(x, decimal.Decimal) else x for x in bindings]
```

Comment:
will this plugin have to handle numbers that double-precision floats can't represent? For example, `2^64` can't be represented by floats. Does pyhive accept `str`s? If so you could use `str` in place of float to get exact representations.

Reply:
I think it's fairly unlikely. If it were to happen, chances are strings would be in use already instead of trying to represent it as a decimal. But not opposed to changing this to `str` if you think it's a better approach.